src/views/TabView/TabView.js

```diff
@@ -110,54 +110,56 @@ class TabView extends PureComponent<void, Props, void> {
 
   _renderTabBar = (props: *) => {
     const {
-      tabBarOptions,
-      tabBarComponent: TabBarComponent,
       animationEnabled,
+      navigation,
+      router,
+      screenProps,
+      tabBarComponent: TabBarComponent,
+      tabBarOptions,
     } = this.props;
     if (typeof TabBarComponent === 'undefined') {
       return null;
     }
+
+    const { state } = navigation;
+    const options = router.getScreenOptions(
+      this.props.childNavigationProps[state.routes[state.index].key],
+      screenProps || {}
+    );
+
+    const tabBarVisible = options.tabBarVisible === null
```

Comment:
Thanks for clarifying, I tested with beta.12 and works fine. Could you tell me with what flow fails?